src/utils/schedule.rs

```diff
@@ -0,0 +1,277 @@
+        let mut start = Tz::UTC.ymd(2022, 10, 25).and_hms(17, 0, 0);
+        let mut end = Tz::UTC.ymd(2022, 10, 25).and_hms(18, 0, 0);
+        assert_eq!(calendar.is_available_between(start, end), true);
+
+        start = Tz::UTC.ymd(2022, 10, 25).and_hms(8, 0, 0);
+        end = Tz::UTC.ymd(2022, 10, 25).and_hms(9, 0, 0);
+        assert_eq!(calendar.is_available_between(start, end), true);
+    }
+
+    #[test]
+    fn test_calendar_with_day_break() {
+        let calendar =
+            Calendar::from_str(&(CAL_WORKDAYS_8AM_6PM.to_owned() + _WITH_1HR_DAILY_BREAK)).unwrap();
+
+        let mut start = Tz::UTC.ymd(2022, 10, 25).and_hms(11, 30, 0);
+        let mut end = Tz::UTC.ymd(2022, 10, 25).and_hms(12, 30, 0);
+        assert_eq!(calendar.is_available_between(start, end), false);
+
+        start = Tz::UTC.ymd(2022, 10, 25).and_hms(8, 0, 0);
+        end = Tz::UTC.ymd(2022, 10, 25).and_hms(12, 0, 0);
+        assert_eq!(calendar.is_available_between(start, end), true);
+
+        start = Tz::UTC.ymd(2022, 10, 25).and_hms(12, 15, 0);
+        end = Tz::UTC.ymd(2022, 10, 25).and_hms(12, 30, 0);
+        assert_eq!(calendar.is_available_between(start, end), false);
+
+        start = Tz::UTC.ymd(2022, 10, 25).and_hms(12, 59, 0);
+        end = Tz::UTC.ymd(2022, 10, 25).and_hms(13, 30, 0);
+        assert_eq!(calendar.is_available_between(start, end), false);
+    }
+
+    #[test]
+    fn test_calendar_with_one_off_block() {
+        let calendar =
+            Calendar::from_str(&(CAL_WORKDAYS_8AM_6PM.to_owned() + _WITH_ONE_OFF_BLOCK)).unwrap();
+
+        let mut start = Tz::UTC.ymd(2022, 10, 26).and_hms(13, 30, 0);
+        let mut end = Tz::UTC.ymd(2022, 10, 26).and_hms(14, 30, 0);
+        assert_eq!(calendar.is_available_between(start, end), false);
+
+        start = Tz::UTC.ymd(2022, 10, 27).and_hms(13, 30, 0);
+        end = Tz::UTC.ymd(2022, 10, 27).and_hms(14, 30, 0);
+        assert_eq!(calendar.is_available_between(start, end), true);
+
+        start = Tz::UTC.ymd(2022, 10, 26).and_hms(11, 00, 0);
+        end = Tz::UTC.ymd(2022, 10, 26).and_hms(13, 30, 0);
+        assert_eq!(calendar.is_available_between(start, end), true);
+
+        start = Tz::UTC.ymd(2022, 10, 26).and_hms(11, 00, 0);
+        end = Tz::UTC.ymd(2022, 10, 26).and_hms(13, 30, 1);
+        assert_eq!(calendar.is_available_between(start, end), false);
+    }
+
+    #[test]
+    fn test_save_and_load_calendar() {
+        let orig_cal_str =
+            &(CAL_WORKDAYS_8AM_6PM.to_owned() + _WITH_1HR_DAILY_BREAK + _WITH_ONE_OFF_BLOCK);
+        let calendar = Calendar::from_str(orig_cal_str).unwrap();
+        let cal_str = calendar.to_string();
+        let calendar = Calendar::from_str(&cal_str).unwrap();
+        assert_eq!(calendar.events.len(), 4);
+        assert_eq!(calendar.events[0].duration, "PT14H");
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_invalid_input() {
+        let calendar = Calendar::from_str(INVALID_CALENDAR).unwrap();
+    }
```

Comment:
test is annotated as `#[should_panic]` so instead of assertion the test is validated by this annotation - test would fail if invalid calendar for some reason didn't panic with error